tests/TestCase/Database/Type/FloatTypeTest.php

```diff
@@ -82,17 +82,36 @@ public function testToPHP()
     {
         $this->assertNull($this->type->toPHP(null, $this->driver));
 
-        $result = $this->type->toPHP('some data', $this->driver);
-        $this->assertSame(0.0, $result);
-
         $result = $this->type->toPHP('2', $this->driver);
         $this->assertSame(2.0, $result);
 
-        $result = $this->type->toPHP('2 bears', $this->driver);
-        $this->assertSame(2.0, $result);
+        $result = $this->type->toPHP('15.3', $this->driver);
+        $this->assertSame(15.3, $result);
+    }
 
-        $result = $this->type->toPHP(['3', '4'], $this->driver);
-        $this->assertSame(1.0, $result);
```

Comment:
The changes here probably require a bit of history:

When I started creating the Type classes, I though that `toPHP` would also be used for marshalling user data. Therefore, I made some checks to avoid warnings when casting bad things. Reality showed that we actually need a `marshal` function, as there was need to have different logic there.

The checks were left there, despite these functions being called almost exclusively with strings, instead of other types. I decided to get rid of those now, so I can take advantage of faster (and less) functions calls.

Reply:
Makes sense. These methods are not exposed to raw user data so they don't need to handle as many scenarios.